conjure_oxide/src/ast.rs

```diff
@@ -39,10 +39,15 @@ pub enum Range<A> {
 }
 
 #[derive(Clone, Debug, PartialEq)]
+#[non_exhaustive]
 pub enum Expression {
     ConstantInt(i32),
     Reference(Name),
     Sum(Vec<Expression>),
     Eq(Box<Expression>, Box<Expression>),
     Geq(Box<Expression>, Box<Expression>),
+
+    // Flattened Constraints
+    SumGeq(Vec<Expression>, Box<Expression>),
```

Comment:
I think this is fine. We want to be able to represent all of the minion input language in our expressions, at least at this stage. A problem that has Eq(Sum,ConstantInt) will be rewritten to SumGeq+SumLeq before getting written to Minion. In fact you probably want `WeightedSumGeq(Vec<(i32, Expression)>, i32)`